File: app/src/main/java/com/example/moviesearcher/view/details/ImageAdapter.java
package com.example.moviesearcher.view.details;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.example.moviesearcher.R;
import com.example.moviesearcher.databinding.ItemImageBinding;

import java.util.ArrayList;
import java.util.List;

public class ImageAdapter extends RecyclerView.Adapter<ImageAdapter.ViewHolder> {

    private List<String> imagePathList = new ArrayList<>();

    public void updateImagePathList(List<String> list){
        imagePathList.clear();
        imagePathList.addAll(list);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        ItemImageBinding view = DataBindingUtil.inflate(inflater, R.layout.item_image, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.itemView.setImagePath(imagePathList.get(position));
    }

    @Override
    public int getItemCount() {
        return imagePathList.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder{

        private ItemImageBinding itemView;

        ViewHolder(@NonNull ItemImageBinding itemView) {
            super(itemView.getRoot());
            this.itemView = itemView;
        }
    }
}

File: app/src/main/java/com/example/moviesearcher/ui/details/media/ImageAdapter.kt
package com.example.moviesearcher.ui.details.media

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.example.moviesearcher.R
import com.example.moviesearcher.databinding.ItemImageBinding
import com.example.moviesearcher.model.data.Image
import java.util.*

class ImageAdapter: RecyclerView.Adapter<ImageAdapter.ViewHolder>(){

    private val imagePathList: MutableList<Image> = ArrayList()

    fun updateImagePathList(list: List<Image>) {
        imagePathList.clear()
        imagePathList.addAll(list)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val view: ItemImageBinding = DataBindingUtil.inflate(inflater, R.layout.item_image, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.onBind(imagePathList[position])
    }

    override fun getItemCount(): Int {
        return imagePathList.size
    }

    inner class ViewHolder(itemView: ItemImageBinding): RecyclerView.ViewHolder(itemView.root){
        private val view = itemView

        fun onBind(image: Image){
            view.image = image
        }
    }
}